### PR TITLE
vo_vaapi_wayland: remove vaapi format query as formats are never used

### PR DESCRIPTION
**Pull request by boxerab**
This code was taken from the older vo_vaapi driver, which does
use the vaapi format list, but the new driver has no use for
these formats, as it is only interested in va surfaces that
can be mapped to wl buffers. The format doesn't enter into
it at all.

### CODE REVIEW

**Comment by boxerab** [comment]
@Dudemanguy  a review por favor  ?

File: video/out/vo_vaapi_wayland.c [hunk 1]
```diff
@@ -266,10 +243,6 @@ static int preinit(struct vo *vo)
         p->display = NULL;
         goto fail;
     }
-    va_get_formats(p);
-    if (!p->image_formats)
-        goto fail;
```
Comment (Dudemanguy) [inline video/out/vo_vaapi_wayland.c:271]: Are you sure this isn't needed? The other `vo_vaapi` driver has this code.

Comment (boxerab) [reply video/out/vo_vaapi_wayland.c:271]: Thanks for the review - I've added more info in the commit and PR description for why this is not needed.

File: video/out/vo_vaapi_wayland.c [hunk 2]
```diff
@@ -266,10 +243,6 @@ static int preinit(struct vo *vo)
...
-
     vo->hwdec_devs = hwdec_devices_create();
```
Comment (Dudemanguy) [inline video/out/vo_vaapi_wayland.c:246]: Could you keep the newline above `vo->hwdec_devs...` too?

File: video/out/vo_vaapi_wayland.c [hunk 3]
```diff
@@ -209,23 +203,6 @@ static struct va_pool_entry *va_pool_alloc_entry(struct vo *vo, struct va_pool *
...
-                                          &formats->num);
-    if (!CHECK_VA_STATUS(ctx, "vaQueryImageFormats()"))
-        return;
-    MP_VERBOSE(ctx, "%d image formats available:\n", formats->num);
-    for (int i = 0; i < formats->num; i++)
-        MP_VERBOSE(ctx, "  %s\n", mp_tag_str(formats->entries[i].fourcc));
-    ctx->image_formats = formats;
-}
-
 static void uninit(struct vo *vo)
```
[the remark this answers is not in the document]
Comment (boxerab) [reply video/out/vo_vaapi_wayland.c:207]: done

File: video/out/vo_vaapi_wayland.c (hunk 2 above)
Comment (boxerab) [reply video/out/vo_vaapi_wayland.c:250]: no problemo

**Comment by boxerab** [comment]
@Dudemanguy do you have any other concerns about this PR ?

**Comment by Dudemanguy** [comment]
Nope, it's good now. Thanks!

**Pull request merged**